Islandora-CLAW/controlled_access_terms
# Fix authority_link JSON-LD serialization (Issue 1259)

**GitHub Issue**: Islandora-CLAW/CLAW/issues/1259

# What does this Pull Request do?

Implements `hook_jsonld_field_mappings()` to tell the JSON-LD serializer that authority_link values should be treated as `xsd:anyURI` just like the URI field it builds on. This fixes the JSON-LD serialization by removing the erroneous `@language` attribute so Fedora won't choke on it.

# What's new?
* Added controlled_access_terms_jsonld_field_mappings() with `authority_link` mapped to `xsd:anyURI`
* Does this change require documentation to be updated? No
* Does this change add any new dependencies? No
* Does this change require any other modifications to be made to the repository
 (ie. Regeneration activity, etc.)? If your controlled access terms failed to index in Fedora before this fix, reindex them once the PR is in place.
* Could this change impact execution of existing code? Fixes it.

# How should this be tested?

- Take an Islandora 8 running Fedora 5.1
- Try to create a Person with an authority link (or any other taxonomy using an authority_link field).
- The term's JSON-LD will include a `@language` property for the schema:sameAs field, you won't find it in Fedora, Milliner will have a 500 error entry, and Karaf will keep spitting out errors until it finally gives up.
- apply the PR to your controlled_access_terms module
- Clear the cache `drush cr`
- Create a new term
- The new term _should_ appear in Fedora.

# Additional Notes:

Fedora 5.1 seems to be less forgiving of our RDF sins. Tread carefully. Also, we mentioned creating a "known issues" page that would address something else that prevents a later version of Fedora without patching; this should go there too.

# Interested parties
@dannylamb, @whikloj, and possibly other @Islandora-CLAW/committers
